core/src/services/icloud/core.rs

```diff
@@ -224,6 +240,19 @@ impl IcloudSigner {
             );
         }
 
+        if !self.data.cookies.is_empty() {
+            let cookies: Vec<String> = self
```

Comment:
We don't need this because reqwest automatically loads and sets `cookies`.

Reply:
Apple server will check 'X-APPLE-WEBAUTH-USER cookie' in cookies.If we don't [update](https://github.com/apache/opendal/pull/4029/files#diff-011a9dd521c8bd6dce3013689fedea6cd090ba73ebc886ab2ed809c2696aa2eeR186) in `authenticate request(setup.icloud.com/setup/ws/1)` to keep value in `seesionData`，the next `sign` cookies are useless.

Reply:
> Apple server will check 'X-APPLE-WEBAUTH-USER cookie' in cookies.If we don't [update](https://github.com/apache/opendal/pull/4029/files#diff-011a9dd521c8bd6dce3013689fedea6cd090ba73ebc886ab2ed809c2696aa2eeR186) in `authenticate request(setup.icloud.com/setup/ws/1)` to keep value in `seesionData`，the next `sign` cookies are useless.

We don't need to. `reqwest` will handle `COOKIES` and `SET-COOKIES` headers exactly the same as you do here.